File: ToDoAngularAspNetCore.Web/ClientApp/src/models/to-do-model.ts
export class ToDoModel {
  id?: number;
  userId: number;
  title: string;
  content: string;
  created: Date;
  isCompleted: boolean
  completedDate?: Date;

  constructor(userId: number, title: string, content: string, created: Date, isCompleted: boolean, completedDate: Date | undefined) {
    this.userId = userId;
    this.title = title;
    this.content = content;
    this.created = created;
    this.isCompleted = isCompleted;
    this.completedDate = completedDate;
  }
}

File: ToDoAngularAspNetCore.Application/Models/ToDoModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ToDoAngularAspNetCore.Application.Models
{
    public class ToDoModel
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public string Title { get; set; }

        public string Content { get; set; }

        public DateTime Created { get; set; }

        public bool IsCompleted { get; set; }

        public DateTime? CompletedDate { get; set; }
    }
}
